### PR TITLE
the prototype for the listener

### PR DESCRIPTION
**Pull request by SimonSaid1996**
push it here just in case anything else happens, the listener will be used, and the publisher is just used for testing

### CODE REVIEW

**Comment by SimonSaid1996** [comment]
please read my shit

**Comment by xinkecf35** [comment]
@SimonSaid1996 Please request pull request reviews from just your respective team. Also, please keep your comments appropriate as everyone in the organization can see this.

File: src/vision/.pytest_cache/v/cache/lastfailed [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (codewizard2) [inline src/vision/.pytest_cache/v/cache/lastfailed]: wtf is this shit, Simon???

Comment (SimonSaid1996) [reply src/vision/.pytest_cache/v/cache/lastfailed]: srry, supposed only to request from the AI team, accidentally send to u

**Pull request merged**